src/intrinsics.cpp

```diff
@@ -625,10 +625,14 @@ static jl_cgval_t generic_bitcast(jl_codectx_t &ctx, ArrayRef<jl_cgval_t> argv)
             vx = ctx.builder.CreateZExt(vx, llvmt);
         } else if (vxt->isPointerTy() && !llvmt->isPointerTy()) {
             vx = ctx.builder.CreatePtrToInt(vx, llvmt);
-            setName(ctx.emission_context, vx, "bitcast_coercion");
+            if (isa<Instruction>(vx) && !vx->hasName())
+                // CreatePtrToInt may undo an IntToPtr
+                setName(ctx.emission_context, vx, "bitcast_coercion");
```

Comment:
Apparently the LLVM IRBuilder is smart enough to collapse `ptrtoint`+`inttoptr` etc into a reference to the original value. If that happens to be an argument, our `setName` aborts. Given that we can't predict whether that would happen, shouldn't we have `setName` bail out for non-Instructions/Constants, instead of aborting and having to check at the call site? We now have many of those already:

```
cgutils.cpp:            if (isa<Instruction>(src) && !src->hasName())
cgutils.cpp:            if (isa<Instruction>(dst) && !dst->hasName())
intrinsics.cpp:            if (isa<Instruction>(vx) && !vx->hasName())
intrinsics.cpp:            if (isa<Instruction>(vx) && !vx->hasName())
intrinsics.cpp:        if (isa<Instruction>(thePtr) && !thePtr->hasName())
```